woocommerce/woocommerce-ios
# [Order Editing] Add edit buttons to shipping and billing addresses

Closes https://github.com/woocommerce/woocommerce-ios/issues/4768.

## Description

This PR adds 2 edit buttons on shipping and billing addresses. They are hidden behind feature flag.

## How

Shipping Address:
- Reused `orderEditingEnabled` from data source to hide changes behind feature flag.
- Added action handler similar to "customer note" edit action.
- In `CustomerInfoTableViewCell` added `editButton` in a UIStackView (since it already used StackView).

Billing Address:
- Added `editingEnabled` init parameter on `BillingInformationViewController` to pass `orderEditingEnabled` flag in `OrderDetailsViewModel`. It won't be enabled if `BillingInformationViewController` initialized from `ReviewOrderViewController`.
- `orderEditingEnabled` property of `OrderDetailsDataSource` is internal to enable access from `OrderDetailsViewModel`.
- To keep edit button in line with other icons sizes on `BillingInformationViewController` - it's implemented with manual constraint manipulations, without StackView. The icon size is 20, but the button is still 44x44 with help of `contentEdgeInsets`. To align 44 button with 20 icons in other cells it has -12 trailing constraint.

## Screenshots

No shipping address|Existing shipping address|Billing address
--|--|--
![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/130619380-22471800-6cd8-4ead-883f-0c368dc1d876.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/130619387-ab753f6d-2f93-4dca-ba87-1fbb2faaefda.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/130619390-0f82fc8c-75fd-46f0-85ad-f9c177db5682.png)

## Test

1. Open existing order
2. Check button on shipping address field, verify it prints to console on tap
3. Tap "View Billing Information"
4. Check button on billing address field, verify it prints to console on tap
5. Go back to order details, tap "Mark Order Complete".
6. Validate that shipping address and billing address (inside billing info subscreen) don't have edit buttons.
7. Disable feature flag and verify that above buttons don't appear.

---
Update release notes:
